[export] Apply move_to_device_pass to all submodules

angelayi opened the pull request:
Previously we only applied this move_to_device_pass to the toplevel graph. However if we have HOO, this pass will not be applied on the HOO submodules. This PR modifies the pass to run on all submodules.

yiming0416 commented on test/export/test_passes.py:
```diff
@@ -1302,6 +1302,28 @@ def forward(self, x):
+
+        ep = torch.export.export(M(), (torch.ones(3),))
+        ep = move_to_device_pass(ep, "cuda")
+        ep.graph_module.submod_1.recompile()
```
Where does the `submod_1` come from?

angelayi replied:
the autocast

yiming0416 commented on torch/export/passes/__init__.py:
```diff
@@ -52,19 +52,21 @@ def _get_new_device(
-            else v,
-            node.meta.get("val"),
-        )
+    for m in ep.graph_module.modules():
```
Can you maybe add some descriptions to the PR summary on when do we need to apply the pass to all submodules?

angelayi committed:
[export] Apply move_to_device_pass to all submodules (#159992)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/159992

Test Plan:
CI

Rollback Plan:

Reviewed By: yiming0416

Differential Revision: D79578038
